components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3429,11 +3449,34 @@ std::string RewardsServiceImpl::GetExternalWalletType() const {
           country_codes::CountryCharsToCountryID(country.at(0), country.at(1));
 
       if (id == current_country)
-        return ledger::constant::kWalletBitflyer;
+        return true;
     }
   }
+  return false;
+}
+
+std::string RewardsServiceImpl::GetExternalWalletType() const {
+  if (IsBitFlyerRegion()) {
+    return ledger::constant::kWalletBitflyer;
+  }
+#if BUILDFLAG(ENABLE_GEMINI_WALLET)
+  if (base::FeatureList::IsEnabled(features::kGeminiFeature)) {
+    return profile_->GetPrefs()->GetString(prefs::kExternalWalletType);
+  }
+#endif
 
   return ledger::constant::kWalletUphold;
```

Comment:
Would be great to have a test for this case

Reply:
Test case here: https://github.com/brave/brave-core/pull/8566/files#diff-79a8ff13d4400b7fb9810d76aa14728acb0aded37d2dc1c414a8b00396489642R118